### ISSUE TITLE
Unused `pandas` dependency in `requirements.txt`

### ISSUE THREAD

**Issue by elcorto** [mala-project/mala#451]
Mala doesn't use pandas

```sh
$ find -name '*.py' | xargs grep pandas
```

Also none of the deps use it:

```sh
$ grep -vE '^\s*$|^#' requirements.txt | xargs -IX sh -c 'pip show X | sed -nre "s/Requires:/X -->/p"'
ase --> matplotlib, numpy, scipy
mpmath -->
numpy -->
optuna --> alembic, cmaes, colorlog, numpy, packaging, PyYAML, sqlalchemy, tqdm
scipy --> numpy
tensorboard --> absl-py, google-auth, google-auth-oauthlib, grpcio, markdown, numpy, protobuf, requests, setuptools, tensorboard-data-server, werkzeug, wheel
```

**Comment by elcorto** [comment]
The mala envs in `install/mala_*.yml` have it also end even pin the version, not sure which package in the conda envs needs it. However, since it is not a dep of mala, I'd remove it from `requirements.txt` at least.

**Comment by RandomDefaultUser** [comment]
Good point, thanks for pointing that out. The problem is that pandas is an optional dependency of optuna, and needed for a few nice functions we use from that library (such as visualization). If pandas is missing, we cannot use the functions, _however_ just installing optuna with the optional dependency flag installs way more then just pandas. Having pandas as a requirement for MALA thus allows us to use the functionalities without installing clutter.
